### PR TITLE
Eng omar ahmed ios gesturesdoc

### PR DESCRIPTION
**Pull request by EngOmarAhmedIOS**
### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by EngOmarAhmedIOS** [comment]
Can anyone review my changes @SSwiniarski @caupolicandiaz

**Comment by EngOmarAhmedIOS** [comment]
nobody reviewed the the file yet, please review it @yoiang @jakemhiller @andrewperk @notalyce

**Comment by SSwiniarski** [comment]
@EngOmarAhmedIOS I'm reviewing it now. Please be patient, the maintainers have a lot of review work to do on this repo, and it can take time to get to things.

**Comment by EngOmarAhmedIOS** [comment]
> @EngOmarAhmedIOS I'm reviewing it now. Please be patient, the maintainers have a lot of review work to do on this repo, and it can take time to get to things.

I am sorry for that, I am just very excited because I want to help other people learn and to docs as many docs as I can.

File: content/swiftui/concepts/gestures/gestures.md
Comment (EngOmarAhmedIOS) [reply content/swiftui/concepts/gestures/gestures.md]: I am facing a problem, I tried to add a gif to the markdown but I don't know if it's working or not. Can you help me solve this problem. Thank you in advance @SSwiniarski

File: content/swiftui/concepts/gestures/gestures.md
Comment (SSwiniarski) [inline content/swiftui/concepts/gestures/gestures.md:36]: @EngOmarAhmedIOS two things about this link. First, rename the image something more meaningful like `swiftui-gestures-example.gif`. Second, the URL you need to use is `https://raw.githubusercontent.com/Codecademy/docs/main/media/` followed by the image name. That path won't render until this is merged, but we want it pointing to the main Docs media folder.

File: content/swiftui/concepts/gestures/gestures.md
Comment (SSwiniarski) [reply content/swiftui/concepts/gestures/gestures.md]: @EngOmarAhmedIOS I left a comment to help you out.

File: content/swiftui/concepts/gestures/gestures.md
Comment (EngOmarAhmedIOS) [reply content/swiftui/concepts/gestures/gestures.md]: I did as you told me, I hope I did it right this time 😅 @SSwiniarski

File: content/swiftui/concepts/gestures/gestures.md
Comment (SSwiniarski) [inline content/swiftui/concepts/gestures/gestures.md:110]: Change the filename on the image, use dashes instead of underscores, and make it all lower case.

Also place some more descriptive text between the `[...]`.

File: content/swiftui/concepts/gestures/gestures.md
Comment (SSwiniarski) [inline content/swiftui/concepts/gestures/gestures.md:139]: Change the filename on the image, use dashes instead of underscores, and make it all lower case.

Also place some more descriptive text between the `[...]`.

File: content/swiftui/concepts/gestures/gestures.md
Comment (SSwiniarski) [inline content/swiftui/concepts/gestures/gestures.md:141]:
```suggestion
the `withAnimation(.spring()){}` was added to gave the text view a spring animation while it returns to its normal size, as it looks in the gif above.
```

File: content/swiftui/concepts/gestures/gestures.md
Comment (SSwiniarski) [inline content/swiftui/concepts/gestures/gestures.md:76]: Correct the spelling on the file name, and, use dashes instead of underscores, and make it all lower case.

Place some more descriptive text between the `[...]`.

File: content/swiftui/concepts/gestures/gestures.md
Comment (SSwiniarski) [inline content/swiftui/concepts/gestures/gestures.md:83]:
```suggestion
```
Remove extra line break.

File: content/swiftui/concepts/gestures/gestures.md
Comment (SSwiniarski) [inline content/swiftui/concepts/gestures/gestures.md:4]:
```suggestion
```
Remove extra line break.

File: content/swiftui/concepts/gestures/gestures.md
Comment (SSwiniarski) [inline content/swiftui/concepts/gestures/gestures.md:39]: Change the file name to use dashes instead of underscores, and make it all lower case.

Also place some more descriptive text between the `[...]`.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@EngOmarAhmedIOS I left a few more comments, mostly around the image filenames, changing them to adhere to our standard filename conventions.

File: media/swiftui-magnificationgesture-example.gif
Comment (SSwiniarski) [inline media/swiftui-magnificationgesture-example.gif:1]: Check the validity of this file, GitHub is not rendering the GIF.

**Review by SSwiniarski — APPROVED** [review]
@EngOmarAhmedIOS Looks good for a second review. However, check the GIF file that isn't rendering in GitHub, it might have been corrupted.

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:13]: Remove all and any comments (these precede with a `#`)

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:16]: Link out to the views topic entry here

Comment (yangc95) [reply content/swiftui/concepts/gestures/gestures.md:16]: Also, place the capitalized `View` in backticks as it represents the actual object, but you can keep the lowercased mentions of view/views as plain formatting.

No pronouns like we, you, I in Docs reference material

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:33]:
```suggestion   
    }
}
```

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:48]:
```suggestion
```

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:52]: add new line between these lines

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:54]:
```suggestion
> **Note:** When a button is needed, a `Button` instance should be used rather than a tap gesture. Any view can be used as the button’s label, and the button type automatically provides many of the standard behaviors that users expect, like accessibility labels and hints.
```

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:67]:
```suggestion
```

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:58]:
```suggestion
`.onLongPressGesture()` is used for handling long press gestures. `.onLongPressGesture()` has an extra parameter called `minimumDuration` which takes a double as the input value. It's used to specify how much time will elapse before an action is executed. For example:
```

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:75]:
```suggestion
```

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:74]:
```suggestion
A string will be printed in the console after pressing the text view for 2 seconds. On Xcode, this will display:

![This is a gif that shows the effect of using LongPressGesture in Xcode simulation](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-longgesture-example.gif)
```

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:79]:
```suggestion
The `gesture()` modifier is used for more advanced and customizable gestures, with one of the gesture `struct`s like: `DragGesture()`, `LongPressGesture()`, `MagnificationGesture()`, `RotationGesture()`, and `TapGesture()`. These all have special modifiers, usually `.onEnded()` and `.onChanged() `, and they can be used to take action when the gestures are in-flight (for `.onChanged()`) or completed (for `.onEnded()`).
```

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:81]:
```suggestion
In the example below, a magnification gesture could be attached to a view so that pinching in and out scales the view up and down. This can be done by creating two `@State` properties to store the scale amount, using that inside a `.scaleEffect()` modifier, then setting those values in the gesture:

```

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:104]:
```suggestion

This will display on Xcode:
```

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:108]:
```suggestion
There is a problem with the code above, because the text view didn't return to its normal state, which can result in a bad user experience. To fix this problem, the `currentAmount` property will be assigned to zero in the `.onEnded()` modifier, so the text view will return to its normal size after completing the gesture: 

```

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:125]:
```suggestion
                            withAnimation(.spring()) {

```

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:133]:
```suggestion

This will display on Xcode:
```

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:39]:
```suggestion
Every time the text view "Hello, World" is tapped, it prints "View Tapped" in the console.

The action can also be implemented after certain amount of tap gestures. For example:
```

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:21]: can you either move this to the same line or add a new line in between?

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:16]: Can you reword this to start with "**Gestures** are..."

File: content/swiftui/concepts/gestures/gestures.md
Comment (yangc95) [inline content/swiftui/concepts/gestures/gestures.md:3]: Terms should starts with a verb, but it's not necessary. I did notice similarities between this and Apple Docs so if we can have this match the introduction that starts with "Gestures are..." that would be 👍

**Review by yangc95 — COMMENTED** [review]
@EngOmarAhmedIOS thanks for this thorough starting entry on gestures in SwiftUI! Great gifs as well, though if you can make the gifs more compact < 1MB so they can be more lightweight and load easily, that would be great.

I left a few more suggestions!

**Comment by EngOmarAhmedIOS** [comment]
you forgot to merge the branch @SSwiniarski @yangc95

**Comment by SSwiniarski** [comment]
@EngOmarAhmedIOS be patient. @yangc95 still needs to review your changes and approve them.

**Comment by SSwiniarski** [comment]
@EngOmarAhmedIOS I did a few small edits to allow the checks to pass, so when @yangc95 does review this again, if she approves, she can merge right away.

**Comment by EngOmarAhmedIOS** [comment]
> @EngOmarAhmedIOS be patient. @yangc95 still needs to review your changes and approve them.

**Comment by EngOmarAhmedIOS** [comment]
Sorry, I did a mistake and closed the pull request by mistake. Sorry

**Review by yangc95 — APPROVED** [review]
@EngOmarAhmedIOS Approving and merging! If the images aren't super clear we can revisit and create a follow-up PR.

**Comment by yangc95** [comment]
🤔 Hmm, what do we think of the images @EngOmarAhmedIOS here's the live link: https://www.codecademy.com/resources/docs/swiftui/gestures

Screenshot 2023-06-02 at 8 26 50 PM

cc: @SSwiniarski

**Comment by EngOmarAhmedIOS** [comment]
> 🤔 Hmm, what do we think of the images @EngOmarAhmedIOS here's the live link: https://www.codecademy.com/resources/docs/swiftui/gestures
> 
> Thanks for this entry! Nicely done
> 
> Screenshot 2023-06-02 at 8 26 50 PM
> cc: @SSwiniarski

I really don’t like how it looks, I think they need to be changed.

**Comment by yangc95** [comment]
@EngOmarAhmedIOS Yeah, I agree. Can you create a PR for edits? I can try to help compress the gifs if you can create a PR with the original files prior to compression

**Comment by EngOmarAhmedIOS** [comment]
> @EngOmarAhmedIOS Yeah, I agree. Can you create a PR for edits? I can try to help compress the gifs if you can create a PR with the original files prior to compression

Okay 👌 I will create them as gif's then I will send them I will make a new PR with the new images. One question what about using videos instead ? @yangc95

**Comment by yangc95** [comment]
@EngOmarElsayed Videos are a possibility, but they are required to go through a more extensive review process. There are some videos up on Docs for some concept entries, but the gifs are easier to access and view

Are you interested in creating videos for SwiftUI?

**Comment by EngOmarElsayed** [comment]
> @EngOmarElsayed Videos are a possibility, but they are required to go through a more extensive review process. There are some videos up on Docs for some concept entries, but the gifs are easier to access and view
> 
> Are you interested in creating videos for SwiftUI?

if you mean that you want me to create videos for all SwiftUI concept, I will be more than happy to help other people learn, I don't mind to go throw a more extensive review process. I am really exited for that.

**Pull request merged**
